### PR TITLE
Avoid marking table to be inserted/deleted as input table

### PR DESCRIPTION
**Pull request by Praveen2112**

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java [hunk 1]
```diff
@@ -451,7 +451,7 @@ protected Scope visitInsert(Insert insert, Optional<Scope> scope)
```
Comment (kokosing) [inline core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java:451]: Can you please first add a commit that reproduces the problem and then with second commit you could fix the problem so we could see how it affects the tests from the first commit?

File: testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java [hunk 2]
```diff
@@ -1048,26 +1067,40 @@ public void testOutputColumnsForSetOperations(String setOperator)
...
 
         runQueryAndWaitForEvents("CREATE VIEW mock.default.create_new_view AS " + baseQuery, 2);
-        assertColumnMetadata(outputColumnMetadata);
+        assertLineage(inputTables, outputColumnMetadata);
 
         runQueryAndWaitForEvents("CREATE VIEW mock.default.create_new_materialized_view AS " + baseQuery, 2);
-        assertColumnMetadata(outputColumnMetadata);
+        assertLineage(inputTables, outputColumnMetadata);
 
         runQueryAndWaitForEvents("INSERT INTO mock.default.table_for_output(test_varchar, test_bigint) " + baseQuery, 2);
```
Comment (kokosing) [inline testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java:1082]: Please also add tests for UPDATE and DELETE (for completeness)

Comment (Praveen2112) [reply testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java:1082]: Applying this change for `UPDATE` and `DELETE` is a bit tricky and can we address them in a different PR ?

File: core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java (hunk 1 above)
Comment (Praveen2112) [reply core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java:451]: Added tests for it.

File: testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java (hunk 2 above)
Comment (kokosing) [reply testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java:1082]: Sure.

**Review by kokosing — APPROVED** [review]

**Review by kokosing — APPROVED** [review]

**Comment by Praveen2112** [comment]
@martint Can you please take a look at this solution.

File: testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java [hunk 3]
```diff
@@ -1087,6 +1087,14 @@ private void assertLineage(String baseQuery, Set<String> inputTables, OutputColu
                         .add("mock.default.table_for_output")
                         .build(),
                 outputColumnMetadata);
+
+        runQueryAndWaitForEvents(format("DELETE FROM mock.default.table_for_output WHERE EXISTS (%s) ", baseQuery), 2);
+        assertLineage(
+                ImmutableSet.<String>builder()
+                        .addAll(inputTables)
+                        // This table shouldn't be a part of the input
+                        .add("mock.default.table_for_output")
```
Comment (martint) [inline testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java]: This seems like a broken test (until the fix in the last commit), so it should be folded into that commit, instead.

**Comment by Praveen2112** [comment]
@martint AC

**Review by martint — APPROVED** [review]

**Pull request merged**